src/llm/io_processing/output_parser.cpp

```diff
@@ -104,7 +105,20 @@ ParsedOutput OutputParser::parse(const std::vector<int64_t>& generatedTokens, co
     return parsedOutput;
 }
 
-std::optional<rapidjson::Document> OutputParser::parseChunk(const std::string& chunkResponse, const bool toolsAvailable) {
+static inline bool isParsingTagPartOfChunk(const std::string& chunk, const std::string& parsingTag) {
+    return chunk.find(parsingTag) != std::string::npos;
+}
```

Comment:
Do we need to wrap it in a separate function since it's one-liner?

Reply:
way cleaner to read actual chunk parser logic